src/TLuaInterpreter.cpp

```diff
@@ -8146,7 +8138,27 @@ int TLuaInterpreter::tempAnsiColorTrigger(lua_State* L)
         }
     }
 
-    int triggerID = pLuaInterpreter->startTempColorTrigger(ansiFgColor, ansiBgColor, code);
+    if (lua_isstring(L, 3)) {
+        code = QString::fromUtf8(lua_tostring(L, 1));
+    } else if (lua_isfunction(L, 3)) {
+        // leave code as a null QString()
+    } else {
+        lua_pushfstring(L, "tempAnsiColorTrigger: bad argument #1 type (code to run as a string or a function expected, got %s!)", luaL_typename(L, 3));
+        return lua_error(L);
+    }
+
+    int expiryCount = -1;
+    if (lua_isnumber(L, 4)) {
```

Comment:
Does this still function if there *isn't* an argument 4? It might do but I can't recall testing it as we've previously done out code to check `lua-gettop(L)` first before doing this test (OTOH we would have been checking that it WASN'T a number and moaning about that rather than silently ignoring a different argument type here)!

Reply:
Yes, it works. I just checked the Lua5.1 source. Values beyond the top-of-stack are definitely `nil`.

Reply:
However if the user gets it wrong and provides something that cannot be *coerced* into a number, like, e.g. a string or the name of a function (an **un**quoted string like `myFunction`) then that will be silently ignored instead of our usual response of firing off a `"tempAnsiColorTrigger: bad argument #4 type (..., got %s!)"` error message and **aborting** the *calling* function (with the `lua_error(lua_State*) function`).